### PR TITLE
Extendable and more customizable `GalleryFragment`

### PR DESCRIPTION
**Pull request by umpteenthdev**

### CODE REVIEW

File: gallery/src/main/java/com/redmadrobot/gallery/ui/GalleryDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        private const val ARG_INITIALLY_CHECKED_ITEM_INDEX = "arg_initially_checked_item_index"
+        private const val ARG_FORCE_ROTATION = "arg_force_rotation"
+        private const val ARG_LAYOUT_RES = "arg_layout_res"
+        private const val ARG_DRAG_LAYOUT_ID = "arg_drag_layout_id"
+        private const val ARG_VIEW_PAGER_ID = "arg_view_pager_id"
+        private const val ARG_SWIPE_TO_DISMISS_LENGTH_DIMEN = "arg_swipe_to_dismiss_length_dimen"
+
+        @JvmOverloads
+        @JvmStatic
+        fun <T : GalleryDialogFragment> T.applyGalleryParams(
```
Comment (terrakok) [inline gallery/src/main/java/com/redmadrobot/gallery/ui/GalleryDialogFragment.kt:165]: Create this method without default parameters, please!
But add this method:
```kotlin
fun create(
  media: ArrayList<Media>,
  position: Int,
  isRotationEnabled: Boolean = false
) : GalleryDialogFragment =
 GalleryDialogFragment().applyGalleryParams(...) //with all default parameters
```

**Comment by terrakok** [comment]
Thank you for contribution! Can you add screenshot to readme with custom design?

**Comment by terrakok** [comment]
@mrblrrd  what do you think?

File: gallery/src/main/java/com/redmadrobot/gallery/ui/GalleryDialogFragment.kt (hunk 1 above)
Comment (umpteenthdev) [reply gallery/src/main/java/com/redmadrobot/gallery/ui/GalleryDialogFragment.kt:165]: Done

**Comment by umpteenthdev** [comment]
> Thank you for contribution! Can you add screenshot to readme with custom design?

Done

**Comment by mrblrrd** [comment]
> @mrblrrd what do you think?

@tehreh1uneh hi! Thank you! I really appreciate you follow the project code style and contribute the clean code. I also encourage your intent to make gallery more flexible. It's a good idea. I think it would be better to develop a system solution for this, not an ad-hoc one. Please give me a couple of days to think about it. Do not hesitate to ping me.

**Comment by umpteenthdev** [comment]
Glad to hear that! Another way, I've considered is to use an inheritance. But I decided to give it up because it would be more difficult for users to extend the base class, also creating custom class constructors is not recommended.
```kotlin
open class GalleryDialogFragment() : DialogFragment() {

    @SuppressLint("ValidFragment")
    constructor(media: ArrayList<Media>,
                position: Int,
                isRotationEnabled: Boolean,
                @DimenRes swipeToDismissLengthDimen: Int,
                @LayoutRes layoutRes: Int,
                @IdRes draggableLayoutId: Int,
                @IdRes viewPagerId: Int): this() {
        if (arguments == null) {
            arguments = Bundle()
        }
        arguments?.apply {
            putSerializable(ARG_LIST_OF_MEDIA, media)
            putInt(ARG_INITIALLY_CHECKED_ITEM_INDEX, position)
            putInt(ARG_LAYOUT_RES, layoutRes)
            putInt(ARG_DRAG_LAYOUT_ID, draggableLayoutId)
            putInt(ARG_VIEW_PAGER_ID, viewPagerId)
            putInt(ARG_SWIPE_TO_DISMISS_LENGTH_DIMEN, swipeToDismissLengthDimen)
            putBoolean(ARG_FORCE_ROTATION, isRotationEnabled)
        }
    }
```
```kotlin
class CustomGalleryDialogFragment : GalleryDialogFragment {

    constructor(): super()

    @SuppressLint("ValidFragment")
    constructor(media: ArrayList<Media>, position: Int): super(
            media = media,
            position = position,
            layoutRes = R.layout.dialog_fragment_custom_gallery,
            viewPagerId = R.id.customIdViewPager,
            draggableLayoutId = R.id.customIdDragLayout,
            isRotationEnabled = true,
            swipeToDismissLengthDimen = R.dimen.default_dismiss_length
    )
```
```kotlin
CustomGalleryDialogFragment(listOfMedia, position)
                            .show(supportFragmentManager, "fragment_tag_custom_gallery")
```

**Pull request closed without merging**